src/components/Dtails.test.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { BrowserRouter } from 'react-router-dom';
+import configureStore from 'redux-mock-store';
```

Comment:
- [ ] Hey, awesome job on setting up those tests! 🧪👏 But it seems like there's a tiny hiccup. 😅 I spotted a couple of issues in your implementation. 🕵️‍♂️

![Screenshot from 2023-09-16 04-48-01](https://github.com/Ness88/Air-Quality-Index-Tunisia/assets/112261237/525b793a-8a55-4875-9cf0-9f55d677187e)


1. It seems you're using Axios, but you forgot to ignore the `node_modules` directory during testing. This might be causing your tests to crash. No worries, though! You can fix it by updating your `package.json` like this:

```json
"scripts": {
  "start": "react-scripts start",
  "build": "react-scripts build",
  "test": "react-scripts test --transformIgnorePatterns \"node_modules/(?!axios)/\"",
  "eject": "react-scripts eject"
}
```

2. It looks like you're using `configureStore` from `redux-mock-store`, but you haven't added it to your `devDependencies`. No problem, you can install it with this command:

```bash
npm install --save-dev redux-mock-store
```

With these friendly suggestions, your tests should run smoothly! 🏃‍♂️💨 Keep up the fantastic work! 🚀🌟